pkg/netpol/connlist/connlist_test.go

```diff
@@ -1,469 +1,559 @@
+			dirName:          filepath.Join("bad_netpols", "subdir2"),
+			errorStrContains: "selector error: key: Invalid value: \"app@b\": name part must consist of alphanumeric characters",
+		},
+		{
+			name:             "Input_dir_has_netpol_with_invalid_rule_peer_should_return_fatal_rule_NetworkPolicyPeer_error",
+			dirName:          filepath.Join("bad_netpols", "subdir3"),
+			errorStrContains: "rule NetworkPolicyPeer error: cannot have both IPBlock and PodSelector/NamespaceSelector set",
+		},
+		{
+			name:             "Input_dir_has_netpol_with_empty_rule_peer_should_return_fatal_rule_NetworkPolicyPeer_error",
+			dirName:          filepath.Join("bad_netpols", "subdir4"),
+			errorStrContains: "rule NetworkPolicyPeer error: cannot have empty rule peer",
+		},
+		{
+			name:             "Input_dir_has_netpol_with_named_port_on_ipblock_peer_should_return_fatal_named_port_error",
+			dirName:          filepath.Join("bad_netpols", "subdir6"),
+			errorStrContains: "named port error: cannot convert named port for an IP destination",
+		},
+		{
+			name:             "Input_dir_does_not_exist_should_return_fatal_error_accessing_directory",
+			dirName:          filepath.Join("bad_yamls", "subdir3"),
+			errorStrContains: "error accessing directory",
+		},
+		{
+			name:    "Input_dir_has_illegal_podlist_pods_with_same_owner_ref_name_has_different_labels_should_return_fatal_error",
+			dirName: "semanticDiff-same-topologies-illegal-podlist",
+			errorStrContains: "Input Pod resources are not supported for connectivity analysis. Found Pods of the same owner demo/cog-agents " +
+				"but with different set of labels.",
 		},
 	}
-
-	for _, entry := range cases {
-		analyzerWithFocusWorkload := NewConnlistAnalyzer(WithFocusWorkload(entry.focusWorkload))
-		dirPath := filepath.Join(testutils.GetTestsDir(), entry.testDirName)
-		res, _, err := analyzerWithFocusWorkload.ConnlistFromDirPath(dirPath)
-		require.Nil(t, err, "test: %s", entry.name)
-		out, err := analyzerWithFocusWorkload.ConnectionsListToString(res)
-		require.Nil(t, err, "test: %s", entry.name)
-		require.Equal(t, entry.expectedConnsOutput, out, "test: %s", entry.name)
-		if entry.notExistingWorkload {
-			// at least one error, one is the warning error on not existing workload
-			caErrors := analyzerWithFocusWorkload.Errors()
-			require.GreaterOrEqual(t, len(caErrors), 1, "test: %s", entry.name)
-			warnType := &connlistAnalyzerWarning{}
-			if len(caErrors) == 1 {
-				require.True(t, errors.As(caErrors[0].Error(), &warnType), "test: %s", entry.name)
-			}
-		}
+	for _, tt := range cases {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			analyzerOpts := []ConnlistAnalyzerOption{WithIncludeJSONManifests()}
+			_, res, err := getConnlistFromDirPathRes(analyzerOpts, tt.dirName)
+			require.Empty(t, res, "test: %q", tt.name)
+			require.Contains(t, err.Error(), tt.errorStrContains, "test: %q", tt.name)
```

Comment:
If there is mismatch in err str, provide a more detailed message, including also info about the actual err str vs the expected one, in addition to the test name.

Reply:
[commit](https://github.com/np-guard/netpol-analyzer/pull/250/commits/7fd2afb3a9c27ba5ba0955ffc13bfc7dfc79c733)